packages/number-field/test/number-field.test.ts

```diff
@@ -943,84 +944,71 @@ describe('NumberField', () => {
             expect(el.value).to.equal(17);
         });
         it('does not select all on click based `focus`', async function () {
-            this.retries(0);
-            const modifier = isMac() ? 'Meta' : 'Control';
             const el = await getElFrom(units({ value: 17 }));
             expect(el.value).to.equal(17);
-            const rect = el.focusElement.getBoundingClientRect();
-            await sendMouse({
-                steps: [
-                    {
-                        type: 'click',
-                        position: [
-                            rect.left + rect.width / 8,
-                            rect.top + rect.height / 2,
-                        ],
-                    },
-                ],
-            });
-            await nextFrame();
-            await nextFrame();
-            await nextFrame();
-            await nextFrame();
+
+            const inputElement = el.shadowRoot?.querySelector(
+                'input'
+            ) as HTMLInputElement;
+            let length: number | null = null;
+
+            await sendMouseTo(el.focusElement, 'click');
             await elementUpdated(el);
             expect(el.focused).to.be.true;
-            await sendKeys({
-                press: `${modifier}+KeyC`,
-            });
-            await nextFrame();
-            await nextFrame();
-            await elementUpdated(el);
-            await sendKeys({
-                press: 'ArrowRight',
-            });
-            await nextFrame();
-            await nextFrame();
-            await elementUpdated(el);
-            await sendKeys({
-                press: `${modifier}+KeyV`,
-            });
-            await nextFrame();
-            await nextFrame();
-            await elementUpdated(el);
-            expect(el.value, 'copy/paste changed the value').to.equal(17);
+
```

Comment:
All we need to do is check the input element selection length once it is ready. No need for all the copying/pasting and `nextFrame()` that are inconsistent.